### ISSUE TITLE
react/prop-types misses props used in an anonymous default-export arrow component

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1234]
## Summary

An anonymous default-export arrow component reads an unvalidated prop but receives no diagnostic. Assigning the same arrow component to a named variable makes both tools report it; a named default-export function also works.

Verified with @utoo/lint 0.4.1, ESLint 8.57.1, @typescript-eslint/parser 6.21.0, eslint-plugin-react 7.37.5, Node.js 22.23.1 on macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react@7.37.5
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "warn"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "warn"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

Run each example separately using:

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Reproduction 1

`example.tsx`:

```tsx
export default (props) => { const { value } = props; return <span>{value}</span>; };
```

```text
ESLint:
'value' is missing in props validation

utlint:
(no diagnostics)
```

## Controls

`example.tsx`:

```tsx
const Example=(props)=>{const{value}=props;return <span>{value}</span>;};export default Example;
```

```text
ESLint:
'value' is missing in props validation

utlint:
'value' is missing in props validation
```

`example.tsx`:

```tsx
export default function Example(props){ return <span>{props.value}</span>; }
```

```text
ESLint:
'value' is missing in props validation

utlint:
'value' is missing in props validation
```

All examples parse successfully. No lint-disable comments or additional rules are involved.
